R/tax_range_strat.R

```diff
@@ -0,0 +1,225 @@
+#'   uncertain (0). By default (\code{certainty = NULL}), no column name is
+#'   provided, and all occurrences are assumed to be certain. In the plot,
+#'   certain occurrences will be plotted with a black circle and joined with
+#'   solid lines, while uncertain occurrences will be plotted with a white
+#'   circle and joined with dashed lines.
+#' @param by \code{character}. How should the output be sorted? Either: "FAD"
+#'   (first appearance; default), "LAD" (last appearance), or "name"
+#'   (alphabetically by taxon names).
+#' @param plot_args A list of optional arguments that are passed directly to
+#'   [graphics::plot()]. Useful arguments include `xlab` (the x-axis label),
+#'   `ylab` (the y-axis label, default is "Bed number"), `main` (the plot
+#'   title), `xlim` (the x-axis limits), and `ylim` (the y-axis limits). The
+#'   `axes` and `type` arguments are not supported and will be overridden.
+#' @param x_args A list of optional arguments that are passed directly to
+#'   [axis()] when generating the x-axis. Useful arguments include `font` (e.g.,
+#'   `3` is italic) and `las` (label orientation).
+#' @param y_args A list of optional arguments that are passed directly to
+#'   [axis()] when generating the y-axis. Useful arguments include `font` (e.g.,
+#'   `3` is italic) and `las` (label orientation).
+#'
+#' @return Invisibly returns a data.frame of the calculated taxonomic
+#'   stratigraphic ranges.
+#'
+#'   The function is usually used for its side effect, which is to create a plot
+#'   showing the stratigraphic ranges of taxa in a section, with levels at which
+#'   the taxon was sampled indicated with a point.
+#'
+#' @section Developer(s): Bethany Allen, William Gearty & Alexander Dunhill
+#' @section Reviewer(s): William Gearty & Lewis A. Jones
+#' @importFrom graphics axis par segments plot points box
+#'
+#' @examples
+#' #Load tetrapod dataset
+#' data(tetrapods)
+#' # Sample tetrapod occurrences
+#' tetrapod_names <- tetrapods$accepted_name[1:50]
+#' # Simulate bed numbers
+#' beds_sampled <- sample.int(n = 10, size = 50, replace = TRUE)
+#' # Simulate certainty values
+#' certainty_sampled <- sample(x = 0:1, size = 50, replace = TRUE)
+#' #Combine into data frame
+#' occdf <- data.frame(taxon = tetrapod_names,
+#'                     bed = beds_sampled,
+#'                     certainty = certainty_sampled)
+#' # Plot stratigraphic ranges
+#' par(mar = c(12, 5, 2, 2))
+#' tax_range_strat(occdf, name = "taxon")
+#' tax_range_strat(occdf, name = "taxon", certainty = "certainty",
+#'                 plot_args = list(ylab = "Stratigraphic height (m)"))
+#' # Plot stratigraphic ranges with more labelling
+#' tax_range_strat(occdf, name = "taxon", certainty = "certainty", by = "name",
+#'                 plot_args = list(main = "Section A",
+#'                                  ylab = "Stratigraphic height (m)"))
+#' eras_custom <- data.frame(name = c("Mesozoic", "Cenozoic"),
```

Comment:
Do you think it is worth having a comment here saying that you can add a geological time scale and that the max_age/min_age refer to the height?

Reply:
No, I think this would be appropriate for the vignette but is too detailed for the simple example here.